pkg/ccl/backupccl/restore_job.go

```diff
@@ -847,7 +848,12 @@ func createImportingDescriptors(
 	// that is, in the 'old' keyspace, before we reassign the table IDs.
 	spans = spansForAllTableIndexes(p.ExecCfg().Codec, tables, nil)
 
-	log.Eventf(ctx, "starting restore for %d mutableTables", len(mutableTables))
+	log.Eventf(ctx, "starting restore for %d tables", len(mutableTables))
+
+	// Assign new IDs to the database descriptors.
+	if err := rewriteDatabaseDescs(mutableDatabases, details.DescriptorRewrites); err != nil {
```

Comment:
was this a bug that this wasn't happening before

Reply:
It was happening before (above). I just moved it to a separate function.